gating/update_dependencies/run

```diff
@@ -1,11 +1,47 @@
-#!/bin/bash -xe
+#!/bin/bash
 
+# Set options to ensure easier debugging
+# and failure on any problems encountered
+set -euxo pipefail
+
+# Find the absolute path to the directory this script
+# is executed from.
 export GATING_PATH="$(readlink -f $(dirname ${0}))"
-source "${GATING_PATH}/../../scripts/functions.sh"
+
+# Set the base RPC-O directory which the functions use
+export BASE_DIR="$(readlink -f ${GATING_PATH}/../..)"
+
+# As get-rpc_release.py requires a yaml import, make
+# sure that it is installed and available.
+if ! python -c 'import yaml' &>/dev/null; then
+  echo "Python YAML library not available. Installing it."
+  apt-get install -y python-yaml
+fi
+
+# Source the functions
+source "${BASE_DIR}/scripts/functions.sh"
+
+# Validate that RPC_RELEASE is set and has a value
+# before continuing. If it is not, then something has
+# gone wrong.
+if [ "${RPC_RELEASE}" == "" ]; then
+  echo "Something has gone wrong: RPC_RELEASE has no value."
+  exit 1
+fi
+
+# We need pip on the host in order to install the required
+# SemVer library, so make sure it is installed and available.
+if ! pip --version &>/dev/null; then
+  echo "Pip not available. Installing it."
+  apt-get install -y python-pip
+fi
 
 # Install the semver library to manipulate the
 # rpc_release version strings
-pip install semver==2.7.9
+if ! python -c 'import semver' &>/dev/null; then
+  echo "Python SemVer library not available. Installing it."
+  pip install semver==2.7.9
```

Comment:
If this requires a specific version the test that it's installed is too loose.

Reply:
Good call - updated to be more specific.